src/TestApplication.cpp

```diff
@@ -2,10 +2,10 @@
 
 TestApplication::TestApplication(int& argc, char** argv)
 : QApplication(argc, argv)
-, serialPort_(new QSerialPort())
 , vehicleData_(new VehicleData())
-, telemetryReporting_(new TelemetryReporting(*serialPort_, *vehicleData_))
-, window_(new Window(*serialPort_, *telemetryReporting_))
+, view_(new View())
+, communicationService_(new CommunicationService(*view_))
+, telemetryReporting_(new TelemetryReporting(communicationService_->getPeripheral(), *vehicleData_, *view_))
```

Comment:
I was thinking of having a window for "Connection" on the UI, it would send a signal that CommunicationService would catch. CS would be able to update it's QIODevice for telemetryReporting. Lemme know your thoughts.